misc/path_utils.c

```diff
@@ -236,6 +237,8 @@ char *mp_normalize_path(void *talloc_ctx, const char *path)
         result = talloc_strndup_append_buffer(result, ptr, next - ptr);
     }
 
+    result = talloc_strdup(talloc_ctx, result);
```

Comment:
I think at this point result is always talloc allocated, so we can use steal trick.
```suggestion
    result = talloc_steal(talloc_ctx, result);
```